File: AdvancedDataStructures_BorislavPanov/mostCommonWord.java
import java.util.HashMap;
import java.util.Map;

public class mostCommonWord {
    public static void main(String[] args) {
        HashMap <String, Integer> map = new HashMap<>();
        Map.Entry<String,Integer> mostCommonWord = null;
        String sentence = "The quick brown fox jumped over the lazy dog. The lazy dog slept all day.";
        sentence = sentence.replaceAll("\\p{Punct}", "");
        sentence = sentence.toLowerCase();
        String[] arraySentence = sentence.split(" ");


        for (String name: arraySentence) {
            if (map.containsKey(name)) {
                map.put(name, map.get(name) + 1);
            } else {
                map.put(name, 1);
            }
        }

        for(Map.Entry<String,Integer> max:map.entrySet())
        {
            if (mostCommonWord == null || max.getValue().compareTo(mostCommonWord.getValue()) > 0)
            {
                mostCommonWord = max;
            }
        }
        System.out.println("\"" + mostCommonWord.getKey() + "\"" + " " + "(appears " + mostCommonWord.getValue() + " times)");
    }
}

File: AdvancedDataStructures_HristoChipev/MostCommonWord.cs
using System;
namespace DictionarysAndMatrix
{
    public class MostCommonWord
    {
        public static string FindMostCommonWord(string str)
        {
            str = str.ToLower();
            var dict = new Dictionary<string, int>();

            foreach (var word in str.Split(" "))
            {
                if (dict.ContainsKey(word))
                {
                    dict[word]++;
                }
                else
                {
                    dict[word] = 1;
                }
            }
            KeyValuePair<string, int> max = dict.First();
            foreach (KeyValuePair<string, int> entry in dict)
            {
                if (max.Value < entry.Value)
                {
                    max = entry;
                }
            }

            return $"'{max.Key}' (appears {max.Value} times)";
        }
    }
}

